scripts/quests/bastok/A_Test_of_True_Love.lua

```diff
@@ -0,0 +1,95 @@
+-----------------------------------
+-- A Test of True Love
+-----------------------------------
+-- Log ID: 1, Quest ID: 62
+-- Carmelo : !gotoid 17743884
+-----------------------------------
+require('scripts/globals/items')
+require('scripts/globals/keyitems')
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.BASTOK, xi.quest.id.bastok.A_TEST_OF_TRUE_LOVE)
+
+quest.reward =
+{
+    fame     = 120,
+    fameArea = xi.quest.fame_area.BASTOK,
+    gil      = 10000,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:hasCompletedQuest(xi.quest.log_id.BASTOK, xi.quest.id.bastok.LOVE_AND_ICE) and
+                player:getFameLevel(xi.quest.fame_area.BASTOK) >= 7 and
+                not player:needToZone() -- need to zone from previous quest Love and Ice
+        end,
+
+        [xi.zone.PORT_BASTOK] =
+        {
+            ['Carmelo'] = quest:progressEvent(270),
+
+            onEventFinish =
+            {
+                [270] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.PORT_BASTOK] =
+        {
+            ['Carmelo'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') < 3 then
+                        return quest:event(271)
+                    elseif quest:getVar(player, 'Prog') == 3 then
+                        return quest:progressEvent(272)
+                    elseif quest:getVar(player, 'Prog') == 4 and player:needToZone() then
+                        return quest:event(273)
+                    elseif quest:getVar(player, 'Prog') == 4 and not player:needToZone() then
+                        return quest:progressEvent(274)
```

Comment:
Not a big deal, but it would be best to throw prog into a local var when it's accessed this many times. Just good code styling and performance nit-pick.